### PR TITLE
DRILL-4152: Add trace logging to  Parquet reader for performance tuning.

### PR DESCRIPTION
**Pull request by parthchandra**
Added trace level logging to track time taken by the Parquet reader.

### CODE REVIEW

File: exec/java-exec/src/main/java/org/apache/drill/exec/store/parquet/ParquetScanBatchCreator.java [hunk 1]
```diff
@@ -122,9 +124,14 @@ public ScanBatch getBatch(FragmentContext context, ParquetRowGroupScan rowGroupS
...
       */
       try {
+        Stopwatch timer = new Stopwatch();
         if ( ! footers.containsKey(e.getPath())){
-          footers.put(e.getPath(),
-              ParquetFileReader.readFooter(conf, new Path(e.getPath())));
+          timer.start();
+          ParquetMetadata footer = ParquetFileReader.readFooter(conf, new Path(e.getPath()));
+          long timeToRead = timer.elapsed(TimeUnit.MICROSECONDS);
+          timer.reset();
```
Comment (adeneche) [inline exec/java-exec/src/main/java/org/apache/drill/exec/store/parquet/ParquetScanBatchCreator.java]: we don't really need to call `timer.reset()` unless we move the timer creation outside the for loop

File: exec/java-exec/src/main/java/org/apache/drill/exec/store/parquet/columnreaders/PageReader.java [hunk 2]
```diff
@@ -124,9 +131,16 @@
...
   private void loadDictionaryIfExists(final ColumnReader<?> parentStatus,
       final ColumnChunkMetaData columnChunkMetaData, final FSDataInputStream f) throws IOException {
+    Stopwatch timer = new Stopwatch();
     if (columnChunkMetaData.getDictionaryPageOffset() > 0) {
       f.seek(columnChunkMetaData.getDictionaryPageOffset());
+      long start=f.getPos();
+      timer.start();
       final PageHeader pageHeader = Util.readPageHeader(f);
+      long timeToRead = timer.elapsed(TimeUnit.MICROSECONDS);
+      timer.reset();
```
Comment (adeneche) [inline exec/java-exec/src/main/java/org/apache/drill/exec/store/parquet/columnreaders/PageReader.java]: no need to call `timer.reset()`

File: exec/java-exec/src/main/java/org/apache/drill/exec/store/parquet/columnreaders/PageReader.java [hunk 3]
```diff
@@ -196,7 +220,15 @@ public boolean next() throws IOException {
...
     do {
+      long start=inputStream.getPos();
+      timer.start();
       pageHeader = dataReader.readPageHeader();
+      long timeToRead = timer.elapsed(TimeUnit.MICROSECONDS);
+      this.updateStats(pageHeader, "Page Header Read", start, timeToRead, 0,0);
+      logger.trace("ParquetTrace,{},{},{},{},{},{},{},{}","Page Header Read","",
+          this.parentColumnReader.parentReader.hadoopPath,
+          this.parentColumnReader.columnDescriptor.toString(), start, 0, 0, timeToRead);
+      timer.reset();
```
Comment (adeneche) [inline exec/java-exec/src/main/java/org/apache/drill/exec/store/parquet/columnreaders/PageReader.java]: same here

File: exec/java-exec/src/main/java/org/apache/drill/exec/store/parquet/columnreaders/ParquetRecordReader.java [hunk 4]
```diff
@@ -105,13 +107,15 @@
   long totalRecordsRead;
   private final FragmentContext fragmentContext;
 
+  public ParquetReaderStats parquetReaderStats = new ParquetReaderStats();
```
Comment (adeneche) [inline exec/java-exec/src/main/java/org/apache/drill/exec/store/parquet/columnreaders/ParquetRecordReader.java]: should we mark this as `final` ? it makes it easier for other developers understand how it's supposed to be used. Of course, we'll have to remove the `set to null` in the `close()` method but I guess that's fine unless of course the close method can be called multiple times

**Comment by adeneche** [comment]
I left some comments that should be easy fixes, other than that LGTM
+1

File: exec/java-exec/src/main/java/org/apache/drill/exec/store/parquet/columnreaders/ParquetRecordReader.java (hunk 4 above)
Comment (parthchandra) [inline exec/java-exec/src/main/java/org/apache/drill/exec/store/parquet/columnreaders/ParquetRecordReader.java]: I did have this as final originally, but close is called multiple times so I had to remove the final.

**Pull request closed without merging**